packages/cuba-react-core/src/data/Collection.tsx

```diff
@@ -72,6 +88,18 @@ export class DataCollectionStore<T> implements DataContainer {
 
   @action
   delete = (e: T & {id?: string}): Promise<any> => {
+    return this.mode === 'client' ? this.deleteClientSide(e) : this.deleteServerSide(e);
```

Comment:
`this.mode === 'client'` used twice in file. May be add getter - later condition could changed so it will be easy to refactor.

Reply:
done